hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/source/TestExpressionEvaluators.java

```diff
@@ -116,11 +124,12 @@ void testEqualTo() {
     assertFalse(equalTo.eval(stats1), "It is not possible to test for NULL values with '=' operator");
   }
 
-  @Test
-  void testNotEqualTo() {
+  @ParameterizedTest
+  @MethodSource("twelveObjects")
+  void testNotEqualTo(Object twelve) {
```

Comment:
Can we write a separate test for all the data types instead of modifying existing test. We just need to trigger the comparison of different data types. For each of the data type, test the comprison of it to all the other data types.

Just a simple `Great Than(GT)` comparision should be enough ~